### PR TITLE
C-sharp-Basic Assignment

### PR DESCRIPTION
**Pull request by Ajinkya066**

### CODE REVIEW

File: C-SharpBasic/CSharp-Assignment-2/StudentData.cs [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+            return Student_Id + "\t " + Student_Name + "\t " +
+                Student_Dept + "\t " + Student_Sem;
+        }
+
+        
+        public void Display()
+        {
+
+           
+            List<StudentData> stu = new List<StudentData>()
```
Comment (poojakalegithub) [inline C-SharpBasic/CSharp-Assignment-2/StudentData.cs:28]: what is stu?

File: C-SharpBasic/CSharp-Assignment-2/ReadFile.cs [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+                    check = sr.ReadLine();
+
+                    while (check != null)
+                    {
+
+                        Console.WriteLine(check);
+                        check = sr.ReadLine();
+                    }
+
+                    sr.Close();
```
Comment (poojakalegithub) [inline C-SharpBasic/CSharp-Assignment-2/ReadFile.cs:28]: What are the standards for closing a connection?

File: C-SharpBasic/CSharp-Assignment-2/ReadFile.cs [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    {
+            String check;
+            public void Read()
+            {
+                try
+                {
+
+                    StreamReader sr = new StreamReader(@"C:\Users\Ajinkya.Sonwane\source\repos\C-Sharp\CSharp-Assignment-2\C# Assignments – II.txt");
+
+                    check = sr.ReadLine();
```
Comment (poojakalegithub) [inline C-SharpBasic/CSharp-Assignment-2/ReadFile.cs:19]: what does check mean?

File: C-SharpBasic/CSharp-Assignment-2/EmployeeSalary.cs [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    {
+        int Emp_Id;
+        string Emp_Name;
+        int Emp_Salary;
+        string Emp_Department;
+
+        public void EmployeeData()
+        {
+            List<EmployeeSalary> employeeSalaries = new List<EmployeeSalary>()
+            {
```
Comment (poojakalegithub) [inline C-SharpBasic/CSharp-Assignment-2/EmployeeSalary.cs:19]: Can we do it in some other way?

File: C-SharpBasic/CSharp-Assignment-2/GenericClass.cs [hunk 5]
```diff
@@ -0,0 +1,30 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace C_Sharp
+{
+    internal class GenericClass<INT, STRING>
```
Comment (poojakalegithub) [inline C-SharpBasic/CSharp-Assignment-2/GenericClass.cs:9]: how generic classes are declared?

**Review by poojakalegithub — CHANGES_REQUESTED** [review]
I have reviewed assignment 1 and 2, just commit the changes as per comments and then you are good to merge this PR.

File: C-SharpBasic/CSharp-Assignment-1/StringOperation.cs [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace C_Sharp
+{
+    class StringOperation
+    {
+        public void Operation()
```
[the remark this answers is not in the document]
Comment (Ajinkya066) [reply C-SharpBasic/CSharp-Assignment-1/StringOperation.cs:11]: fixed it. new name is stringOperation.

File: C-SharpBasic/CSharp-Assignment-1/StringOperation.cs [hunk 7]
```diff
@@ -0,0 +1,39 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace C_Sharp
+{
+    class StringOperation
```
[the remark this answers is not in the document]
Comment (Ajinkya066) [reply C-SharpBasic/CSharp-Assignment-1/StringOperation.cs:9]: Internal will be default modifier if we not specified.

File: C-SharpBasic/CSharp-Assignment-1/StringOperation.cs [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+    {
+        public void Operation()
+        {
+
+            String name = "Ajinkya";
+            String surname = "Sonawane";
+            Console.Write("Enter a String : ");
+            string Name = Console.ReadLine();
+            char[] stringArray = Name.ToCharArray();
```
[the remark this answers is not in the document]
Comment (Ajinkya066) [reply C-SharpBasic/CSharp-Assignment-1/StringOperation.cs:18]: fixed it. new name is "UserName" as performing action names.

File: C-SharpBasic/CSharp-Assignment-1/StringOperation.cs [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+            Array.Reverse(stringArray);
+            string reverseString = new string(stringArray);
+
+            Console.WriteLine("First name = {0}", name);
```
[the remark this answers is not in the document]
Comment (Ajinkya066) [reply C-SharpBasic/CSharp-Assignment-1/StringOperation.cs:22]: Yes, by using ${variable} which called interpolated string.

File: C-SharpBasic/CSharp-Assignment-1/Enum.cs [hunk 10]
```diff
@@ -0,0 +1,18 @@
...
+using System.Text;
+using System.Threading.Tasks;
+
+namespace C_Sharp
+{
+    internal class Enum
+    {
+        public enum Countries
+        {
+            INDIA = 1, AUSTRALIA, USA, JAPAN, FRANCE, ENGLAND
```
[the remark this answers is not in the document]
Comment (Ajinkya066) [reply C-SharpBasic/CSharp-Assignment-1/Enum.cs:13]: I added functionality in program class main method.

File: C-SharpBasic/CSharp-Assignment-1/Inheritance.cs [hunk 11]
```diff
@@ -0,0 +1,110 @@
...
+        }
+
+        public void Run()
+        {
+            int P = 20000, R = 8, T = 1;
+            int SI;
+            String FirstName = "Ajinkya";
+            String LastName = "Sonawane";
+            String city = "Dhule";
+
```
[the remark this answers is not in the document]
Comment (Ajinkya066) [reply C-SharpBasic/CSharp-Assignment-1/Inheritance.cs:32]: User input added.

File: C-SharpBasic/CSharp-Assignment-2/StudentData.cs (hunk 1 above)
Comment (Ajinkya066) [reply C-SharpBasic/CSharp-Assignment-2/StudentData.cs:28]: object of list StudentData. I change the name to "student".

File: C-SharpBasic/CSharp-Assignment-2/ReadFile.cs (hunk 3 above)
Comment (Ajinkya066) [reply C-SharpBasic/CSharp-Assignment-2/ReadFile.cs:19]: It contains text data from file in string format. name is change is stringContainer.

File: C-SharpBasic/CSharp-Assignment-2/GenericClass.cs (hunk 5 above)
Comment (Ajinkya066) [reply C-SharpBasic/CSharp-Assignment-2/GenericClass.cs:9]: In C# generic classes are defined using type parameter brackets after class name.

File: C-SharpBasic/CSharp-Assignment-2/ReadFile.cs (hunk 2 above)
Comment (Ajinkya066) [reply C-SharpBasic/CSharp-Assignment-2/ReadFile.cs:28]: We should close the connection in finally block as it will run regardless of output.
fixed it.

**Review by poojakalegithub — APPROVED** [review]

**Pull request merged**